onnx/helper.py

```diff
@@ -1599,9 +1598,38 @@ def np_dtype_to_tensor_dtype(np_dtype: np.dtype) -> int:
     Returns:
         TensorsProto's data_type
     """
-    return cast(
-        int,
-        mapping._NP_TYPE_TO_TENSOR_TYPE[np_dtype],
+    if np_dtype in mapping._NP_TYPE_TO_TENSOR_TYPE:
+        return cast(
+            int,
+            mapping._NP_TYPE_TO_TENSOR_TYPE[np_dtype],
+        )
+
+    if np.issubdtype(np_dtype, np.str_):
+        return TensorProto.STRING
+
+    if np_dtype == custom_np_types.float8e4m3fn and np_dtype.descr[0][0] == "e4m3fn":
```

Comment:
Just trying to understand: why is the second check above required?

Reply:
I tried without and it failed. numpy does not seem to check the description, only the underlying type (uint8 here).